### PR TITLE
Make log rotation configurable

### PR DESCRIPTION
**Pull request by vegardengen**
Adding possibility to configure more how many logs to keep, or to disable deletion of all logs alltogether, with the new config option maxlogfiles (0 = disable deleting old files)

Adding maxlogfilesize (in MB)

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -157,6 +158,7 @@ type config struct {
 	ReadMacPath    string `long:"readonlymacaroonpath" description:"Path to write the read-only macaroon for lnd's RPC and REST services if it doesn't exist"`
 	InvoiceMacPath string `long:"invoicemacaroonpath" description:"Path to the invoice-only macaroon for lnd's RPC and REST services if it doesn't exist"`
 	LogDir         string `long:"logdir" description:"Directory to log output."`
+	MaxLogFiles    int64  `long:"maxlogfiles" description:"Maximum logfiles to keep (0 for no rotation)"`
```
Comment (halseth) [inline config.go]: Maybe just make this `Int`?

Comment (halseth) [reply config.go]: Maybe also add a comment how big each file will be?

Comment (vegardengen) [reply config.go]: Make log file size configurable instead?

**Comment by vegardengen** [comment]
Squashed, rebased, formatted. Issues fixed, I think?

**Pull request merged**